src/app/quick_views/media.cpp

```diff
@@ -0,0 +1,226 @@
+        QFont myFont("arial", 10);
+        painter.setFont(myFont);
+        painter.setClipRect(height(),0, text_rect_width ,height());
+        painter.drawText(title_pos, track_info);
+        painter.setClipping(false);
+    }
+
+    /*** PREV BUTTON ***/
+    QIcon prev_icon(new IconEngine(this->arbiter, QString("://icons/skip_previous.svg"), true));
+    QPixmap prev_pixmap = prev_icon.pixmap(QSize(128,128));
+    painter.drawPixmap(prev_rect,prev_pixmap);
+
+    /*** PLAY/PAUSE BUTTON ***/
+    if(this->playback_.has_track_progress() && connected &&
+                metadata_.has_track_length() && metadata_.track_length() != 0){
+            
+        QPen progressBGPen = QPen(arbiter.theme().color(),4);
+        painter.setPen(progressBGPen);
+        painter.drawArc(play_rect,1440,5760);
+        QPen progressPen = QPen(QColor(70,70,70),4);
+        painter.setPen(progressPen);  
+        int step =  5760 / metadata_.track_length();
+        int invVal = metadata_.track_length() - playback_.track_progress();
+        int val = step * invVal;
+        painter.drawArc(play_rect,1440,val);
+        
+    } else {
+        QPen progressPen = QPen(QColor(70,70,70),4);
+        painter.setPen(progressPen);  
+        painter.drawArc(play_rect,1440,5760);
+    }
+
+    if(playback_.has_playback_state() && playback_.playback_state() == aasdk::proto::messages::MediaInfoChannelPlaybackData_PlaybackState_PLAY){
+        QIcon icon(new IconEngine(this->arbiter, QString("://icons/pause.svg"), true));
+        QPixmap pause = icon.pixmap(QSize(128,128));
+        painter.drawPixmap(play_rect,pause);
+       
+    } else {
+        QIcon icon(new IconEngine(this->arbiter, QString("://icons/play.svg"), true));
+        QPixmap play = icon.pixmap(QSize(128,128));
+        painter.drawPixmap(play_rect,play);
+    }
+
+    /**** NEXT BUTTON ****/
+    QIcon next_icon(new IconEngine(this->arbiter, QString("://icons/skip_next.svg"), true));
+    QPixmap next_pixmap = next_icon.pixmap(QSize(128,128));
+    painter.drawPixmap(next_rect,next_pixmap);
+
+}
+
+void MediaWidget::resizeEvent(QResizeEvent *event)
+{
+    prev_rect = QRect((width() / 2) - (height() / 2 + height()), 8,height() - 16,height() - 16);
+    play_rect = QRect((width() / 2) - (height() / 2), 8,height() - 16,height() - 16);
+    next_rect = QRect((width() / 2) + (height() / 2), 8,height() - 16,height() - 16);
+
+}
+
+void MediaWidget::mousePressEvent(QMouseEvent *event)
+{
+
+    if(event->button()==Qt::LeftButton)
+    {
+
+        AAHandler *aa_handler = arbiter.android_auto().handler;
+        if(prev_rect.contains(event->pos())){
+            aa_handler->injectButtonPress(aasdk::proto::enums::ButtonCode::PREV,openauto::projection::ButtonEventType::RELEASE);
+        }
+        if(play_rect.contains(event->pos())){
+            aa_handler->injectButtonPress(aasdk::proto::enums::ButtonCode::TOGGLE_PLAY,openauto::projection::ButtonEventType::PRESS);
```

Comment:
I thought I had made a helper method that sent both press and release events for you, but I can't find where that ended up (if it made it in at all), but might be something to double check

Reply:
there's the AAHandler::injectButtonPressHelper, is that what you're talking about?

Reply:
Mmm, no, that's for the event filter stuff.